Index: src/main/java/dao/CityDAOImpl.java
===================================================================
diff --git a/src/main/java/dao/CityDAOImpl.java b/src/main/java/dao/CityDAOImpl.java
--- a/src/main/java/dao/CityDAOImpl.java	
+++ b/src/main/java/dao/CityDAOImpl.java	
@@ -14,6 +14,7 @@
             Transaction transaction = session.beginTransaction();
             session.save(city);
             transaction.commit();
+            session.close();
         }
     }
     @Override
